pyproject.toml

```
@@ -28,6 +28,12 @@ matplotlib = "^3.9.2"
 pylint = "^3.3.1"
 notebook = "^7.2.2"
 
+[tool.poetry.requires-plugins]
+poetry-dynamic-versioning = { version = ">=1.0.0,<2.0.0", extras = ["plugin"] }
+
+[tool.poetry-dynamic-versioning]
+enable = true
+
```

Consider adding a substitution area for PDV where it can supply a version number for reference within the package itself. This is useful for quickly checking which version of a package is installed; e.g. `comparison_tools.__version__` from an addition in the `__init__.py`: `__version__ = "0.0.0"`

```suggestion

[tool.poetry-dynamic-versioning.substitution]
files = [ "src/comparison_tools/__init__.py" ]
```

---

Added!